src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -350,36 +347,35 @@ public OpenIdConnectHandler(HttpClient backchannel)
                 {
                     return null;
                 }
+                message = messageReceivedContext.ProtocolMessage;
 
-                var properties = new AuthenticationProperties();
-
-                // if state is missing, just log it
-                if (string.IsNullOrEmpty(message.State))
+                // if any of the error fields are set, throw error null
+                if (!string.IsNullOrEmpty(message.Error))
```

Comment:
Consider moving this block after the `state` check to reject requests missing a valid `state` since it must be returned by the server even on errored responses:

http://openid.net/specs/openid-connect-core-1_0.html#AuthError
https://tools.ietf.org/html/rfc6749#section-4.1.2.1